jobs/spFilteringJob.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/data-preservation-programs/spade-tenant/config"
+	"github.com/data-preservation-programs/spade-tenant/db"
+	"github.com/joho/godotenv"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"gorm.io/gorm"
+)
+
+type SP struct {
+	Version        int       `bson:"__v"`
+	Provider       string    `bson:"_id"`
+	AgentCity      string    `bson:"agentCity"`
+	AgentCountry   string    `bson:"agentCountry"`
+	AgentLatitude  float64   `bson:"agentLatitude"`
+	AgentLongitude float64   `bson:"agentLongitude"`
+	AgentRegion    string    `bson:"agentRegion"`
+	Date           time.Time `bson:"date"`
+	LatencyMs      float64   `bson:"latencyMs"`
+	Multiaddr      string    `bson:"multiaddr"`
+	TestId         string    `bson:"testId"`
+}
+
+func main() {
+
+	sps := setUpEnvAndFetchSpDataFromMongo()
+
+	config := config.InitConfig()
+	database, err := db.OpenDatabase(config.DB_URL, config.DEBUG, config.DRY_RUN)
+
+	if err != nil {
+		panic(err)
+	}
+
+	var clauses []db.TenantSPEligibilityClauses
+	database.Model(&db.TenantSPEligibilityClauses{}).Find(&clauses)
+	tenantClauseMap := createTenantClauseMap(clauses)
+
+	for tenantId, clauses := range tenantClauseMap {
+		for _, sp := range sps {
+			isValid := isValidClause(clauses, sp)
+			updateDatabase(isValid, database, sp, tenantId)
+		}
+	}
+}
+
+func setUpEnvAndFetchSpDataFromMongo() []SP {
+	godotenv.Load()
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
+	opts := options.Client().ApplyURI(os.Getenv("MONGODB_URI")).SetServerAPIOptions(serverAPI)
+
+	client, err := mongo.Connect(context.TODO(), opts)
```

Comment:
make this function take a `ctx context.Context` from the caller, then pass that in to this `mongo.Connect()` function. Instantiate the `ctx := context.TODO()` at the top level of `main()` instead